pkg/component/gardenercontrollermanager/gardener_controller_manager_test.go

```diff
@@ -0,0 +1,471 @@
+				Name:      managedResourceNameVirtual,
+				Namespace: namespace,
+			},
+		}
+		managedResourceSecretRuntime = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "managedresource-" + managedResourceRuntime.Name,
+				Namespace: namespace,
+			},
+		}
+		managedResourceSecretVirtual = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "managedresource-" + managedResourceVirtual.Name,
+				Namespace: namespace,
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		deployer = New(fakeClient, namespace, fakeSecretManager, values)
+	})
+
+	Describe("#Deploy", func() {
+		Context("resources generation", func() {
+			BeforeEach(func() {
+				Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(managedResourceRuntime), managedResourceRuntime)).To(BeNotFoundError())
+				Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(managedResourceVirtual), managedResourceVirtual)).To(BeNotFoundError())
+				Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(managedResourceSecretRuntime), managedResourceSecretRuntime)).To(BeNotFoundError())
+				Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(managedResourceSecretVirtual), managedResourceSecretVirtual)).To(BeNotFoundError())
+
+				Expect(fakeClient.Create(ctx, &resourcesv1alpha1.ManagedResource{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       managedResourceNameRuntime,
+						Namespace:  namespace,
+						Generation: 1,
+					},
+					Status: healthyManagedResourceStatus,
+				})).To(Succeed())
+
+				Expect(fakeClient.Create(ctx, &resourcesv1alpha1.ManagedResource{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       managedResourceNameVirtual,
+						Namespace:  namespace,
+						Generation: 1,
+					},
+					Status: healthyManagedResourceStatus,
+				})).To(Succeed())
+			})
+
+			It("should successfully deploy all resources", func() {
+				Expect(deployer.Deploy(ctx)).To(Succeed())
+
+				Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(managedResourceRuntime), managedResourceRuntime)).To(Succeed())
+				Expect(managedResourceRuntime).To(Equal(&resourcesv1alpha1.ManagedResource{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: resourcesv1alpha1.SchemeGroupVersion.String(),
+						Kind:       "ManagedResource",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:            managedResourceRuntime.Name,
+						Namespace:       managedResourceRuntime.Namespace,
+						ResourceVersion: "2",
+						Generation:      1,
+						Labels:          map[string]string{"gardener.cloud/role": "seed-system-component"},
+					},
+					Spec: resourcesv1alpha1.ManagedResourceSpec{
+						Class:       pointer.String("seed"),
+						SecretRefs:  []corev1.LocalObjectReference{{Name: managedResourceSecretRuntime.Name}},
+						KeepObjects: pointer.Bool(false),
+					},
+					Status: healthyManagedResourceStatus,
+				}))
+
+				Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(managedResourceSecretRuntime), managedResourceSecretRuntime)).To(Succeed())
```

Comment:
Might need some adaptations like https://github.com/gardener/gardener/blob/410ba4938bcaf7b6d4aa1b8119d784e20eb73193/pkg/component/gardenerscheduler/gardener_scheduler_test.go#L339-L340 after rebasing on current master.
Similarly for the other MR secret.

Reply:
Thanks for the hint.
Indeed, it required the changes from https://github.com/gardener/gardener/pull/8116 after rebasing.

Will be included in the next commit.